libmate-desktop/mate-colorsel.c

```diff
@@ -2239,25 +2222,28 @@ palette_change_notify_instance (GObject    *object,
 }
 
 static void
-default_noscreen_change_palette_func (const GdkColor *colors,
+default_noscreen_change_palette_func (const GdkRGBA *colors,
 				      gint            n_colors)
 {
   default_change_palette_func (gdk_screen_get_default (), colors, n_colors);
 }
 
 static void
 default_change_palette_func (GdkScreen	    *screen,
-			     const GdkColor *colors,
+                             const GdkRGBA  *colors,
                              gint            n_colors)
 {
+  GValue value = G_VALUE_INIT;
   gchar *str;
 
+  g_value_init (&value, G_TYPE_STRING);
+
   str = mate_color_selection_palette_to_string (colors, n_colors);
+  g_value_set_string (&value, str);
 
-  gtk_settings_set_string_property (gtk_settings_get_for_screen (screen),
+  g_object_set_property (G_OBJECT (gtk_settings_get_for_screen (screen)),
```

Comment:
Any reason not to use `g_object_set()` instead?  I didn't check but I believe you leak the string set into `value`, and although that's easily fixable `g_object_set()` is a lot easier to use in C:
```c
g_object_set (gtk_settings_get_for_screen (screen), "gtk-color-palette", str, NULL);
```

No need for messing with `GValue`s manually.

Reply:
done